packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -60,9 +60,13 @@ export function extractLogSafeErrorProperties(error: any, sanitizeStack: boolean
 /** type guard for ILoggingError interface */
 export const isILoggingError = (x: any): x is ILoggingError => typeof x?.getTelemetryProperties === "function";
 
-/** Copy props from source onto target, overwriting any keys that are already set on target */
-function copyProps(target: unknown, source: ITelemetryProperties) {
-    Object.assign(target, source);
+/** Copy props from source onto target, but do not overwrite an existing prop that matches */
+function copyProps(target: ITelemetryProperties | LoggingError, source: ITelemetryProperties) {
```

Comment:
the `| LoggingError` blows my mind! i thought we got totally away from just setting props on objects

Reply:
Nope I dropped that PR from Vlad's pushback.  Instead just merged a PR that added the omitKeys stuff.  I see pros/cons to both sides, although now we're in a worse world where one error implementation sets the props on the object (LoggingError) but the other doesn't (SimpleFluidError).  That's been making me feel uneasy so I'll open an issue on it.

Reply:
https://github.com/microsoft/FluidFramework/issues/7018